test/testdata/rewriter/minitest.rb.rewrite-tree.exp

```diff
@@ -14,11 +14,41 @@ class <emptyTree><<C <root>>> < ()
     end
 
     begin
-      ::T::Sig::WithoutRuntime.sig() do ||
-        <self>.params({}).void()
+      <emptyTree>::<C CONST> = ::T.unsafe(nil)
+      begin
+        ::T::Sig::WithoutRuntime.sig() do ||
+          <self>.params({}).void()
+        end
+        def <test_allows constants inside of IT><<C <todo sym>>>(&<blk>)
+          ::Module.const_set(:"CONST", 10)
+        end
+      end
+    end
+
+    begin
+      <emptyTree>::<C C2> = ::T.let(::T.unsafe(nil), <emptyTree>::<C Integer>)
+      begin
+        ::T::Sig::WithoutRuntime.sig() do ||
+          <self>.params({}).void()
+        end
+        def <test_allows let-ed constants inside of IT><<C <todo sym>>>(&<blk>)
+          ::Module.const_set(:"C2", <emptyTree>::<C T>.let(10, <emptyTree>::<C Integer>))
+        end
```

Comment:
This shows that constant definitions with a `T.let` preserve their type annotation when pulled out.